demo/site/src/app/not-found.tsx

```diff
@@ -0,0 +1,12 @@
+import Link from "next/link";
+
+export default async function NotFound404(): Promise<JSX.Element> {
```

Comment:
```suggestion
export default function NotFound404() {
```

This type can be inferred